sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1805,9 +1805,6 @@ class Dataset[T] private[sql](
    * @since 1.6.0
    */
   def sample(withReplacement: Boolean, fraction: Double, seed: Long): Dataset[T] = {
-    require(fraction >= 0,
```

Comment:
I don't think you can remove this, but you can improve the condition to limit to <= 1 when without replacement